vault/resource_pki_secret_backend_role.go

```diff
@@ -631,8 +659,10 @@ func pkiSecretBackendRoleUpdate(d *schema.ResourceData, meta interface{}) error
 		data["ext_key_usage"] = extKeyUsage
 	}
 
-	if len(policyIdentifiers) > 0 {
-		data["policy_identifiers"] = policyIdentifiers
+	if len(policyIdentifiersList) > 0 {
+		data["policy_identifiers"] = policyIdentifiersList
+	} else if policyIdentifierBlocks != "" {
+		data["policy_identifiers"] = policyIdentifierBlocks
```

Comment:
Same suggestion as in  the `Read()` method. Ideally we would merge the two and rely on `d.HasChange()`, but I think that is probably out of scope for this PR.

```suggestion
	if v, ok := d.GetOk("policy_identifiers"); ok {
		data["policy_identifiers"] = v
	} else if v, ok := d.GetOk("policy_identifier"); ok {
		data["policy_identifiers"] = internal.ReadPolicyIdentifierBlocks(v.(*schema.Set))
	}
```